WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -926,6 +930,12 @@ private void handleOpenPageIntent(@NonNull Intent intent) {
                 case ARG_BLOGGING_PROMPTS_ONBOARDING:
                     showBloggingPromptsOnboarding();
                     break;
+                case ARG_MEDIA:
+                    if (!mSelectedSiteRepository.hasSelectedSite()) {
+                        initSelectedSite();
+                    }
+                    mActivityNavigator.viewCurrentBlogMedia(this, getSelectedSite());
+                    break;
```

Comment:
Hey @zwarm 

I updated logic to use `ActivityNavigator` here. I think we should gradually move all the Activity Navigation logic to `ActivityNavigator`. Commit a0d27d189db58f21f4aa5be3896247917f87ce37

Reply:
Great idea - I forget `ActivityNavigator` existed.